roxctl/central/db/generate/generate.go

```diff
@@ -0,0 +1,171 @@
+	centralDBCertGeneratePath = "/api/extensions/certgen/centraldb"
+)
+
+var (
+	errCreateCentralDBBundleSupported = errox.InvariantViolation.New("central does not support central db bundle functionality")
+	centralDBCertBundle               = set.NewFrozenStringSet(mtls.CACertFileName, mtls.CentralDBCertFileName, mtls.CentralDBKeyFileName)
+)
+
+type generateCommand struct {
+	// Properties that are bound to cobra flags.
+	config *renderer.Config
+
+	// Properties that are injected or constructed.
+	env environment.Environment
+
+	// timeout to make Central API call
+	timeout time.Duration
+}
+
+// Command represents the generate command.
+func Command(cliEnvironment environment.Environment) *cobra.Command {
+	cmd := &generateCommand{config: &cfg, env: cliEnvironment}
+
+	c := &cobra.Command{
+		Use:    "generate",
+		Hidden: true,
+	}
+
+	if !buildinfo.ReleaseBuild {
+		flags.AddHelmChartDebugSetting(c)
+	}
+	c.PersistentFlags().BoolVar(&cmd.config.EnablePodSecurityPolicies, "enable-pod-security-policies", true, "Create PodSecurityPolicy resources (for pre-v1.25 Kubernetes)")
+	c.PersistentPreRunE = func(*cobra.Command, []string) error {
+		cmd.construct(c)
+		return cmd.populateMTLS()
+	}
+
+	c.AddCommand(k8s(cliEnvironment))
+	c.AddCommand(openshift(cliEnvironment))
+
+	return c
+}
+
+func (cmd *generateCommand) populateMTLS() error {
+	logger := cmd.env.Logger()
+	logger.InfofLn("Populating Central DB Certificate from bundle...")
+	fileMap, err := zipdownload.GetZipFiles(zipdownload.GetZipOptions{
+		Path:       centralDBCertGeneratePath,
+		Method:     http.MethodPost,
+		Timeout:    cmd.timeout,
+		BundleType: "central-db",
+		ExpandZip:  true,
+	}, logger)
+	if err != nil {
+		return err
+	}
+	err = verifyCentralDBBundleFiles(fileMap)
+	if err != nil {
+		return err
+	}
+	cmd.config.SecretsByteMap = map[string][]byte{
+		"ca.pem":              fileMap[mtls.CACertFileName].Content,
+		"central-db-cert.pem": fileMap[mtls.CentralDBCertFileName].Content,
+		"central-db-key.pem":  fileMap[mtls.CentralDBKeyFileName].Content,
+		"central-db-password": []byte(renderer.CreatePassword()),
+	}
+	return nil
+}
+
+func (cmd *generateCommand) construct(c *cobra.Command) {
+	cmd.timeout = flags.Timeout(c)
+}
+
+func generateBundleWrapper(config renderer.Config) (*zip.Wrapper, error) {
+	rendered, err := render(config)
+	if err != nil {
+		return nil, err
+	}
+
+	wrapper := zip.NewWrapper()
+	wrapper.AddFiles(rendered...)
+	return wrapper, errors.Wrap(err, "could not get scanner bundle")
+}
+
+func outputZip(logger logger.Logger, config renderer.Config) error {
+	logger.InfofLn("Generating Central DB bundle...")
+	common.LogInfoPsp(logger, config.EnablePodSecurityPolicies)
+	wrapper, err := generateBundleWrapper(config)
+	if err != nil {
+		return err
+	}
+	var outputPath string
+	if roxctl.InMainImage() {
+		bytes, err := wrapper.Zip()
+		if err != nil {
+			return errors.Wrap(err, "error generating zip file")
+		}
+		_, err = os.Stdout.Write(bytes)
```

Comment:
We made a huge effort to not use `os` in roxctl. Please use logger instead. With `os.Stdout` there is no way to unit test it. Please consider adding unit tests.